crates/node/core/src/version.rs

```diff
@@ -82,9 +80,53 @@ pub fn default_extra_data_bytes() -> Bytes {
 /// The default client version accessing the database.
 pub fn default_client_version() -> ClientVersion {
     ClientVersion {
-        version: CARGO_PKG_VERSION.to_string(),
-        git_sha: VERGEN_GIT_SHA.to_string(),
-        build_timestamp: VERGEN_BUILD_TIMESTAMP.to_string(),
+        version: env!("CARGO_PKG_VERSION").to_string(),
+        git_sha: env!("VERGEN_GIT_SHA_SHORT").to_string(),
+        build_timestamp: env!("VERGEN_BUILD_TIMESTAMP").to_string(),
+    }
+}
+
+/// Global static version metadata
+static VERSION_METADATA: OnceLock<RethCliVersionConsts<'static>> = OnceLock::new();
+
+/// Initialize the global version metadata.
+pub fn init_version_metadata(metadata: RethCliVersionConsts<'static>) {
+    let _ = VERSION_METADATA.set(metadata);
+}
```

Comment:
we can name this `try_int_version_metadata` and return the error as well, also move top of file